### PR TITLE
Middleware: Add team metadata to HTTP handlers

### LINKED ISSUE #67521: Add team label to `grafana_http_request` metrics to make it possible to have different SLO's per team.
By added `team` as an label to the HTTP metrics we could define different SLO's for different teams and route alerts based on these labels. I'm using `team` because its the best word to describe who owns it but I'm very open for a better name. 

ex
```
grafana_http_request_duration_seconds_bucket{team="core", handler="/",method="GET",status_code="200",le="0.005"} 0
grafana_http_request_duration_seconds_bucket{team="core", handler="/",method="GET",status_code="200",le="0.01"} 0
```

```
grafana_http_request_duration_seconds_bucket{team="alerting", handler="/alerts/test",method="GET",status_code="200",le="0.005"} 0
grafana_http_request_duration_seconds_bucket{team="alerting", handler="/alerts/test",method="GET",status_code="200",le="0.01"} 0
```

We should default to `core` or something similar. 
I think this could be implemented as an `web.Handler` similar to how we add the handler name. https://github.com/grafana/grafana/blob/main/pkg/middleware/request_tracing.go#L28

This would probably only be interesting for us as an organisation so putting it behind a setting would make sense since it could be confusing for other operators.

### PR DESCRIPTION
**Pull request by bergquist**
This PR adds an team label to the `grafana_http_request_duration_seconds_.*` which will make it easy to track error rate per HTTP handler and route alerts to the right team. 

closes https://github.com/grafana/grafana/issues/67521

### CODE REVIEW

**Comment by bergquist** [comment]
I want to reconsider the naming a bit before anyone reviews this.

**Comment by bergquist** [comment]
ready for review! I'll make sure to add tests before merging this. I just didn't have time to write those before going on PTO :)

**Comment by ryantxu** [comment]
Is team the codeowner?  Or service owner?

**Comment by bergquist** [comment]
Service owner/on call schedule. 

This label would allow us to calculate the error rate for different teams and send alerts to them instead of passing everything thru the same channel. It would also be possible for each team to track their own SLO.

**Comment by bergquist** [comment]
ping @grafana/backend-platform @gotjosh 

This should make it easier to track error rate per teams instead of matching different handlers. 

I only added the middleware to a few alerting handlers to prove the point. Ideally the alerting team can do some follow up work to add it for all relevant handlers.

**Review by zserge — APPROVED** [review]

File: pkg/middleware/requestmeta/request_metadata.go
Comment (tolzhabayev) [inline pkg/middleware/requestmeta/request_metadata.go:10]: maybe it makes sense to somehow unify those lists?

Comment (tolzhabayev) [reply pkg/middleware/requestmeta/request_metadata.go:10]: with https://github.com/grafana/grafana/blob/8ec4c1bdc87c43fe43e6bc56db924a9107edf137/pkg/services/featuremgmt/codeowners.go

Comment (bergquist) [reply pkg/middleware/requestmeta/request_metadata.go:10]: I was thinking about that but codeowners are not really the same since these will be used to route alerts to on call schedules and multiple code owners might be part of one schedule. 

Long term it might make sense to move both these consts and the code owner consts to some kind of code meta package. But lets do that later :)

Comment (danielthedm) [reply pkg/middleware/requestmeta/request_metadata.go:10]: +1 that's what I came here to say. We should abstracts teams to a single enum that we attach escalation policies various metadata to that we can reuse across packages for whatever the use case might be

**Pull request merged**
